src/rest.cpp

```diff
@@ -265,6 +267,206 @@ static bool rest_block_notxdetails(HTTPRequest* req, const std::string& strURIPa
     return rest_block(req, strURIPart, false);
 }
 
+static bool rest_filter_header(HTTPRequest* req, const std::string& strURIPart)
+{
+    if (!CheckWarmup(req))
+        return false;
+    std::string param;
+    const RetFormat rf = ParseDataFormat(param, strURIPart);
+
+    std::vector<std::string> uriParts;
+    boost::split(uriParts, param, boost::is_any_of("/"));
+    if (uriParts.size() != 3) {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid URI format. Expected /rest/blockfilterheaders/filtertype/count/blockhash");
+    }
+
+    uint256 block_hash;
+    if (!ParseHashStr(uriParts[2], block_hash)) {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Invalid hash: " + uriParts[2]);
+    }
+
+    BlockFilterType filtertype;
+    if (!BlockFilterTypeByName(uriParts[0], filtertype)) {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Unknown filtertype " + uriParts[0]);
+    }
+
+    BlockFilterIndex* index = GetBlockFilterIndex(filtertype);
+    if (!index) {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Index is not enabled for filtertype " + uriParts[0]);
+    }
+
+    long count = strtol(uriParts[1].c_str(), nullptr, 10);
+    if (count < 1 || count > 2000) {
+        return RESTERR(req, HTTP_BAD_REQUEST, "Header count out of range: " + uriParts[1]);
```

Comment:
I think it would be useful if this error message provided the user with the valid range.

Reply:
Hmm, its the same limits as for headers (the code was copied from that section above). Not sure where to document this given the lack of equivalent documentation like we have in RPC where we can fail with help text. Maybe we should have some kind of much more informative general help system like we do for RPC, though its obviously out of scope here.

Reply:
similarly, it'd make sense to define a constant for this, and use it in both places, instead of hardcoding 2000